trin-core/src/portalnet/storage.rs

```diff
@@ -362,28 +362,11 @@ impl PortalStorage {
         Ok(size)
     }
 
-    /// Internal method that returns the distance between our node ID and a given content ID
+    /// Internal method that returns the distance between our node ID and a given content ID.
+    /// Returns the most significant 8 bytes of the distance as a u64.
     fn distance_to_content_id(&self, content_id: &[u8; 32]) -> u64 {
-        // naked unwrap since content_id and node_id are equally sized
-        let byte_vector = xor_two_values(content_id, &self.node_id.raw().to_vec()).unwrap();
-
-        PortalStorage::byte_vector_to_u64(byte_vector)
-    }
-
-    // Takes the most significant 8 bytes of a vector and casts them into a u64.
-    // With a 32-byte big-endian unsigned integer input, this is equivalent to dividing by 2^(256 - 32)
-    fn byte_vector_to_u64(vec: Vec<u8>) -> u64 {
-        if vec.len() < 8 {
-            debug!("Error: XOR returned less than 8 bytes.");
-            return 0;
-        }
-
-        let mut array: [u8; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
-        for (index, byte) in vec.iter().take(8).enumerate() {
-            array[index] = byte.clone();
-        }
-
-        u64::from_be_bytes(array)
+        let distance = xor(content_id, &self.node_id.raw());
+        distance.0[3]
```

Comment:
I'm probably confused here, my byte math/knowledge is fairly shallow, I'm sure I'm misunderstanding something. Doesn't big endian mean the most significant bytes are stored first (AIU in the left-most slots => `distance.0[0]`?) - but we are returning the least significant (furthest to the right => `distance.0[3]`) 64 bits (or the 4th of 4x8 byte arrays in the `U256`)?

Reply:
Yes, but the [U256](https://docs.rs/ethereum-types/0.12.1/ethereum_types/struct.U256.html#) stores the data in little endian internally.

Reply:
Ahh gotcha, good to know